File: CS/App_Code/BootstrapThemeModel.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Xml;
using System.Xml.Serialization;

/// <summary>
/// Summary description for ThemeModel
/// </summary>
[XmlRoot("Themes")]
public class BootstrapThemesModel {
    const string ResourcePath= "SwitcherResources\\Themes.xml";
    static BootstrapThemesModel current;
    static readonly object currentLock = new object();
    public static BootstrapThemesModel Current {
        get {
            lock(currentLock) {
                if(current == null) {
                    using(FileStream fileStream = new FileStream(AppDomain.CurrentDomain.BaseDirectory + "\\" + ResourcePath, FileMode.Open)) {
                        XmlReader reader = XmlReader.Create(fileStream);
                        XmlSerializer serializer = new XmlSerializer(typeof(BootstrapThemesModel));
                        current = (BootstrapThemesModel)serializer.Deserialize(reader);
                        fileStream.Close();
                    }
                }
                return current;
            }
        }
    }
    List<ThemeModelBase> _themes = new List<ThemeModelBase>();
    [XmlElement("Theme")]
    public List<ThemeModelBase> Themes {
        get { return _themes; }
    }
}

File: VB/App_Code/BootstrapThemeModel.vb
Imports System
Imports System.Collections.Generic
Imports System.IO
Imports System.Xml
Imports System.Xml.Serialization

''' <summary>
''' Summary description for ThemeModel
''' </summary>
<XmlRoot("Themes")> _
Public Class BootstrapThemesModel
    Private Const ResourcePath As String= "SwitcherResources\Themes.xml"

    Private Shared current_Renamed As BootstrapThemesModel
    Private Shared ReadOnly currentLock As New Object()
    Public Shared ReadOnly Property Current() As BootstrapThemesModel
        Get
            SyncLock currentLock
                If current_Renamed Is Nothing Then
                    Using fileStream As New FileStream(AppDomain.CurrentDomain.BaseDirectory & "\" & ResourcePath, FileMode.Open)
                        Dim reader As XmlReader = XmlReader.Create(fileStream)
                        Dim serializer As New XmlSerializer(GetType(BootstrapThemesModel))
                        current_Renamed = DirectCast(serializer.Deserialize(reader), BootstrapThemesModel)
                        fileStream.Close()
                    End Using
                End If
                Return current_Renamed
            End SyncLock
        End Get
    End Property
    Private _themes As New List(Of ThemeModelBase)()
    <XmlElement("Theme")> _
    Public ReadOnly Property Themes() As List(Of ThemeModelBase)
        Get
            Return _themes
        End Get
    End Property
End Class